src/lib.rs

```diff
@@ -97,11 +97,11 @@ impl Client {
             .build()
             .map_err(|e| ClientError::Other(e.to_string()))?;
         let base_url = base_parsed;
-        let auth_token = Self::create_token(&client, &base_url, username, password).await?;
         Ok(Client {
             client,
             base_url,
-            auth_token,
+            // must call login() to request a new auth token
+            auth_token: "".to_owned(),
```

Comment:
If we do it this way (rather than with the distinct client types proposed in another comment) then we should make `auth_token` an Option rather than having a magic empty string value.

Reply:
I think I'm leaning more towards the explicit types idea you had. The idea you brought up about accidentally passing credentials when we don't want to is a *great* idea.